test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -259,6 +259,7 @@ public static void resetPortCounter() {
         // TODO: consolidate logging initialization for tests so it all occurs in logconfigurator
         LogConfigurator.loadLog4jPlugins();
         LogConfigurator.configureESLogging();
+        MockLogAppender.init();
```

Comment:
This needs to be called every time we (re) configure log4j in tests, right?
So besides the special case above, ESTestCase (should) cover all, correct?

Reply:
Yep, any tests subclassing LuceneTestCase directly will need to call it if they use MockLogAppender.